Enterprise-CMCS/eAPD
# 4366 - Date validation is inconsistent between inline and admin check(backend)

Resolves #4366 

### Description
Date validation is happening between the in-line and admin check is inconsistent. When the admin check is enabled the validation in the page appears to work as expected. The admin check panel:

- Doesn't catch when Start Dates are later than End Dates
- Doesn't validate when an invalid year is entered

### Chromatic Link
https://www.chromatic.com/build?appId=61d5b948cf6f17003a12bf77&number=1335

### Significant changes or possible side effects
Per discussion, valid years are 1960 thru 2151.

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| Date field with start and end date | admin check is on | should catch if end date is before start date with appropriate validation error | cypress |
| Date field | admin check is one | should catch an invalid year with appropriate validation error | cypress |

### Steps to manually verify this change

1. Login and view an APD. Turn on Admin Check
2. Navigate to Activity Schedule
3. Enter dates

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when QA has


### This pull request can be merged when
